### PR TITLE
Added new ADR for BackstopJS

### PR DESCRIPTION
**Pull request by alexjanousekGSA**
ADR outlining proposal for adding backstop JS to admin project to improve QA. 

Original Issue including screenshots done during testing can be found [here](https://github.com/GSA/notifications-admin/issues/1876)

Details around the deep dive and POC can be found [here](https://docs.google.com/document/d/1K2FxD4iCXjOCSFE9kott6kchtBE7ziMkmYr1LifPKw4/edit)

### CODE REVIEW

File: docs/adrs/0009-adr-implement-backstopjs-to-improve-qa.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Status: Proposed
+Date: August 27th, 2024
+
+### Context
+We're looking to integrate BackstopJS, a visual regression testing tool, into our Admin UI project to improve QA and keep our UI consistent. This tool will help catch visual bugs early and make sure our design stays on track. We considered several options: deferring the integration, minimal integration with our current tools, full integration using Docker, and an optional testing setup. The goal is to find a balance between ease of use for developers and thorough testing while making sure the integration fits well with our current CI/CD pipeline.
+
+### Decision
+We decided to integrate BackstopJS as an optional part of our workflow. This means developers can run visual regression tests when they think it's needed, using specific Gulp commands. By doing this, we keep the process flexible and less friction for those who are new to the tool. We’ll also provide clear documentation and training to help everyone get up to speed. If this approach works well, we might look into making these tests a regular part of our process down the road by integrating into the CI/CD process.
```
Comment (ccostino) [inline docs/adrs/0009-adr-implement-backstopjs-to-improve-qa.md:10]: One small grammatical edit!

```suggestion
We decided to integrate BackstopJS as an optional part of our workflow. This means developers can run visual regression tests when they think it's needed, using specific Gulp commands. By doing this, we keep the process flexible and minimize friction for those who are new to the tool. We’ll also provide clear documentation and training to help everyone get up to speed. If this approach works well, we might look into making these tests a regular part of our process down the road by integrating into the CI/CD process.

```

**Review by ccostino — COMMENTED** [review]
Awesome, thanks @alexjanousekGSA!

This looks good to me aside from one minor grammatical change.  I'll give folks a bit of time to weigh in as well. 🙂

**Review by A-Shumway42 — APPROVED** [review]
Approved. Looking forward to working with BackstopJS!

**Review by heyitsmebev — APPROVED** [review]
Thanks @alexjanousekGSA for solutions to testing! I don't have any questions about BackstopJS for now. I am curious by the potential of BackstopJS. I'd like to see BackstopJS as an option for testing until we become more familiar with it and understand its use cases better. We already have many different testing tools on the frontend, and while BackstopJS is not exactly the same, managing and testing with multiple tools can be cumbersome.

That said, I'm curious about the tool and do think it could be helpful to see at a glance which parts of the site have changed. This could be particularly useful if an image goes missing unexpectedly or if there are CSS or layout changes. I'm open to exploring the value and insights it could provide.

It would be helpful to run it quickly, similar to how we use flake8 or isort, but overall, I think we should keep it optional for now.

**Review by xlorepdarkhelm — APPROVED** [review]

**Review by ccostino — APPROVED** [review]
Thanks for writing this up, @alexjanousekGSA! 🎉 

Once this is merged, we're going to make a few adjustments and additions to it, and flip it to the `accepted` state.

**Pull request merged**